frontend/appflowy_flutter/lib/workspace/presentation/settings/widgets/settings_language_view.dart

```diff
@@ -12,85 +14,86 @@ class SettingsLanguageView extends StatelessWidget {
+        onPressed: () {},
+      ),
+      popupBuilder: (BuildContext context) {
+        final allLocales = EasyLocalization.of(context)!.supportedLocales;
 
-class _LanguageSelectorDropdownState extends State<LanguageSelectorDropdown> {
-  Color currHoverColor = Colors.white.withOpacity(0.0);
-  void hoverExitLanguage() {
-    setState(() {
-      currHoverColor = Colors.white.withOpacity(0.0);
-    });
+        return ConstrainedBox(
+          constraints: const BoxConstraints(maxHeight: 400),
+          child: ListView.builder(
+            itemBuilder: (context, index) {
+              final locale = allLocales[index];
+              return LanguageItem(locale: locale, currentLocale: currentLocale);
+            },
+            itemCount: allLocales.length,
+          ),
+        );
+      },
+    );
   }
+}
 
-  void hoverEnterLanguage() {
-    setState(() {
-      currHoverColor = Theme.of(context).colorScheme.secondaryContainer;
-    });
-  }
+class LanguageItem extends StatelessWidget {
+  final Locale locale;
+  final Locale currentLocale;
+  const LanguageItem({
+    super.key,
+    required this.locale,
+    required this.currentLocale,
+  });
 
   @override
   Widget build(BuildContext context) {
-    return MouseRegion(
-      onEnter: (_) => hoverEnterLanguage(),
-      onExit: (_) => hoverExitLanguage(),
-      child: Container(
-        margin: const EdgeInsets.symmetric(horizontal: 8),
-        decoration: BoxDecoration(
-          borderRadius: BorderRadius.circular(8),
-          color: currHoverColor,
-        ),
-        child: DropdownButtonHideUnderline(
-          child: Padding(
-            padding: const EdgeInsets.symmetric(horizontal: 6),
-            child: DropdownButton<Locale>(
-              value: context.locale,
-              dropdownColor: Theme.of(context).cardColor,
-              onChanged: (locale) {
-                context
-                    .read<AppearanceSettingsCubit>()
-                    .setLocale(context, locale!);
-              },
-              autofocus: true,
-              borderRadius: BorderRadius.circular(8),
-              items:
-                  EasyLocalization.of(context)!.supportedLocales.map((locale) {
-                return DropdownMenuItem<Locale>(
-                  value: locale,
-                  child: Padding(
-                    padding: const EdgeInsets.all(12.0),
-                    child: FlowyText.medium(
-                      languageFromLocale(locale),
-                      color: Theme.of(context).colorScheme.tertiary,
-                    ),
-                  ),
-                );
-              }).toList(),
-            ),
-          ),
+    return SizedBox(
+      height: 32,
+      child: FlowyButton(
+        text: FlowyText.medium(
+          languageFromLocale(locale),
         ),
+        rightIcon: currentLocale == locale
+            ? const FlowySvg(name: 'grid/checkmark')
+            : const SizedBox(),
```

Comment:
`rightIcon` can be null if I remember correctly, so don't need to return a widget.
```suggestion
            : null,
```